pyne/mcnp.py

```diff
@@ -1990,210 +1987,235 @@ def _read_meshtal_head(self, f):
+        res_tot : numpy array
+            Tally total flux result.
+        rel_err_tot : numpy array
+            Total relative error fo flux.
+        """
 
         # get result and relative error data from file
-        result = np.empty(shape=(num_e_groups, num_vol_elements))
-        rel_error = np.empty(shape=(num_e_groups, num_vol_elements))
+        result = np.empty(shape=(num_e_groups, num_ves))
+        rel_error = np.empty(shape=(num_e_groups, num_ves))
         for i in range(0, num_e_groups):
             result_row = []
             rel_error_row = []
-            for j in range(0, num_vol_elements):
+            for j in range(0, num_ves):
                 line = f.readline().split()
-                result_row.append(float(line[self._column_idx["Result"]]))
+                result_row.append(float(line[column_idx["Result"]]))
                 rel_error_row.append(
-                    float(line[self._column_idx["Rel_Error"]]))
-
+                    float(line[column_idx["Rel_Error"]]))
             result[i] = result_row
             rel_error[i] = rel_error_row
 
-        # Tag results and error vector to mesh
-        self.tag(self.tag_names[0], tagtype='nat_mesh',
-                 size=num_e_groups, dtype=float)
-        res_tag = self.get_tag(self.tag_names[0])
-        self.tag(self.tag_names[1], tagtype='nat_mesh',
-                 size=num_e_groups, dtype=float)
-        rel_err_tag = self.get_tag(self.tag_names[1])
-
-        if num_e_groups == 1:
-            res_tag[:] = result[0]
-            rel_err_tag[:] = rel_error[0]
-        else:
-            res_tag[:] = result.transpose()
-            rel_err_tag[:] = rel_error.transpose()
-
+        res_tot = []
+        rel_err_tot = []
         # If "total" data exists (i.e. if there is more than
         # 1 energy group) get it and tag it onto the mesh.
         if num_e_groups > 1:
-            result = []
-            rel_error = []
-            for i in range(0, num_vol_elements):
+            for i in range(0, num_ves):
                 line = f.readline().split()
-                result.append(float(line[self._column_idx["Result"]]))
-                rel_error.append(
-                    float(line[self._column_idx["Rel_Error"]]))
-
-            self.tag(self.tag_names[2], size=1,
-                     dtype=float, tagtype='nat_mesh')
-            res_tot_tag = self.get_tag(self.tag_names[2])
-
-            self.tag(self.tag_names[3], size=1,
-                     dtype=float, tagtype='nat_mesh')
-            rel_err_tot_tag = self.get_tag(self.tag_names[3])
-
-            res_tot_tag[:] = result
-            rel_err_tot_tag[:] = rel_error
+                res_tot.append(float(line[column_idx["Result"]]))
+                rel_err_tot.append(
+                    float(line[column_idx["Rel_Error"]]))
+        else:
+            res_tot = result.flatten()
+            rel_err_tot = rel_error.flatten()
```

Comment:
It looks like these lines could cause an error if `num_e_groups > 1`.

`res_tot` and `rel_err_tot` are initialized as empty lists. If we get this point we'll get an `AttributeError` when calling `flatten` I think.

Reply:
Well, it is the `result` that calls `flatten()`, but not `res_tot`.
The `result` is initialized by `result = np.empty(shape=(num_e_groups, num_ves))` and then filled with data.

Reply:
Ah, so true. I definitely didn’t read that correctly. Thanks!

Can we ensure that the `res_tot` and `res_err_tot` containers are consistent? In one case they will be lists, and in the other they will be NumPy arrays. Unless I’m missing something else!

Reply:
They are convert to `np.array` before return in https://github.com/pyne/pyne/pull/1199/commits/c8c6cdc1ece48782d7ec5a096de2bc3c069c97f0.